test/CMakeLists.txt

```diff
@@ -185,7 +185,7 @@ else()
         ${PROJECT_SOURCE_DIR}/test/src/mbgl/test/http_server.cpp
         PROPERTIES
         COMPILE_FLAGS
-        -Wno-shadow
+        $<$<NOT:$<BOOL:${WIN32}>>:-Wno-shadow>
```

Comment:
Would 

```
$<$<OR:$<CXX_COMPILER_ID:GNU>,$<CXX_COMPILER_ID:Clang>>:-Wno-shadow>
```

be more appropriate here?

Reply:
I agree, this is better. Maybe even better `$<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-shadow>`, since only Visual Studio doen't support it?

Reply:
What other compilers support it? I would just stick to the ones that support it.

Reply:
I searched about it and I agree with you, better stick with the explicit compiler list to enable this flag.